### ISSUE TITLE
scale_color.* not working correctly

### ISSUE THREAD

**Issue by Deleetdk** [tengfei/ggbio#79]
I'm trying to dynamically color some regions of the genome using `layout_circle`. I am able to map the values to the data, but I am unable to modify the color scale to suit my purposes. Trying to do so results in a duplicated color scale with 2 guides. But the color mapping stays the way it is. The plot below shows an example.

![ggbio_color_scale_bug](https://cloud.githubusercontent.com/assets/6759913/18058671/abc351f6-6e15-11e6-8070-db749d90ab23.png)

We see that the colored regions are mapped to the default blueish scale, while my substitute scale has a guide but is not used. Oddly, the names of the guides match despite it not working.

The code for the plot is:

```
#load data
hg19Ideo = readRDS("hg19Ideo.RDS")
genome_ranges = readRDS("genome_ranges.RDS")
#load library and plot
library(ggbio)
ggplot() +
      layout_circle(hg19Ideo, 
                    geom = "ideo", 
                    fill = "gray70", 
                    radius = 25,
                    trackWidth = 5) +
      layout_circle(hg19Ideo,
                    geom = "text", 
                    aes(label = seqnames), 
                    vjust = 0,
                    radius = 37, 
                    trackWidth = 7, 
                    size = 4, 
                    angle = v_text_angles) +
      layout_circle(genome_ranges,
                    geom = "rect",
                    radius = 20,
                    trackWidth = 4,
                    aes = aes(color = color_score,
                              fill = color_score)
                    ) +
      scale_color_gradient(low = "green", high = "red", limits = c(0, 100))
```

I have tried other `scale_color_*` functions, such as those given as example in the ggbio book, with no luck.

Is this a bug or am I doing something wrong?

The data used for the plot is attached.

[data.tar.gz](https://github.com/tengfei/ggbio/files/442885/data.tar.gz)

**Comment by lawremi** [comment]
You need to also specify the scale for the fill (in addition to the stroke color), i.e., add this scale:

```
scale_fill_gradient(low = "green", high = "red", limits = c(0, 100))
```

**Comment by Deleetdk** [comment]
Doh! My mistake!

**Comment by Deleetdk** [comment]
Actually, it still does not work correctly.

```
    ggplot() +
      layout_circle(hg19Ideo, 
                    geom = "ideo", 
                    fill = "gray70", 
                    radius = 25,
                    trackWidth = 5) +
      layout_circle(hg19Ideo,
                    geom = "text", 
                    aes(label = seqnames), 
                    vjust = 0,
                    radius = 37, 
                    trackWidth = 7, 
                    size = 4, 
                    angle = v_text_angles) +
      layout_circle(hg19Ideo,
                    geom = "rect",
                    fill = "green",
                    color = "green",
                    radius = 20,
                    trackWidth = 4) +
      layout_circle(genome_ranges,
                    geom = "rect",
                    radius = 20,
                    trackWidth = 4,
                    aes = aes(color = score, fill = score)) #+
```

Two errors:
1. Only one guide is created, while two should have been (one for fill, one for color).
2. The color scale (peripheral line) is black despite being sat to aes to match the score variable. It's hard to see on the small image. The second picture zooms in.

![color_fill](https://cloud.githubusercontent.com/assets/6759913/18190961/b9514c02-70c6-11e6-8230-fc27f3e26fb5.png)
![zoom_genome](https://cloud.githubusercontent.com/assets/6759913/18190989/f5fdfdbc-70c6-11e6-82c0-b2861e914687.png)

If we ignore this and try to set the color scales appropriately:

    ggplot() +
      layout_circle(hg19Ideo, 
                    geom = "ideo", 
                    fill = "gray70", 
                    radius = 25,
                    trackWidth = 5) +
      layout_circle(hg19Ideo,
                    geom = "text", 
                    aes(label = seqnames), 
                    vjust = 0,
                    radius = 37, 
                    trackWidth = 7, 
                    size = 4, 
                    angle = v_text_angles) +
      layout_circle(hg19Ideo,
                    geom = "rect",
                    fill = "green",
                    color = "green",
                    radius = 20,
                    trackWidth = 4) +
      layout_circle(genome_ranges,
                    geom = "rect",
                    radius = 20,
                    trackWidth = 4,
                    aes = aes(color = score, fill = score)) +
      scale_color_gradient(low = "green", high = "red", limits = c(0, 100), guide = F) +
      scale_fill_gradient(low = "green", high = "red", limits = c(0, 100), guide = F)

This fixes the fill scale, but the color scale stays the same.

![color_broken](https://cloud.githubusercontent.com/assets/6759913/18191181/96f904b8-70c8-11e6-9aa6-4c1cfab7fa08.png)

So, it seems there is a hidden default `color = "black"` that overwrites the `aes(color = score)` call. We can test this by explicitly setting `color = NA`, i.e.:


    ggplot() +
      layout_circle(hg19Ideo, 
                    geom = "ideo", 
                    fill = "gray70", 
                    radius = 25,
                    trackWidth = 5) +
      layout_circle(hg19Ideo,
                    geom = "text", 
                    aes(label = seqnames), 
                    vjust = 0,
                    radius = 37, 
                    trackWidth = 7, 
                    size = 4, 
                    angle = v_text_angles) +
      layout_circle(hg19Ideo,
                    geom = "rect",
                    fill = "green",
                    color = "green",
                    radius = 20,
                    trackWidth = 4) +
      layout_circle(genome_ranges,
                    geom = "rect",
                    radius = 20,
                    trackWidth = 4,
                    color = NA, #overwrite hidden default
                    aes = aes(color = score, fill = score)) +
      scale_color_gradient(low = "green", high = "red", limits = c(0, 100), guide = F) +
      scale_fill_gradient(low = "green", high = "red", limits = c(0, 100), guide = F)

Then we get the correct result for this plot.

![correct](https://cloud.githubusercontent.com/assets/6759913/18191270/6c5eb7e2-70c9-11e6-9181-a3a3a9b88f07.png)

However, there is still a problem. Here's another datafile. 
[data.tar.gz](https://github.com/tengfei/ggbio/files/451081/data.tar.gz)
The second datafile results in this plot:

![color_stays_green](https://cloud.githubusercontent.com/assets/6759913/18191350/ed648cf4-70c9-11e6-94f1-ae0c4dd6e6b9.png)
![color_stays_green](https://cloud.githubusercontent.com/assets/6759913/18191388/2b684ea0-70ca-11e6-9f5a-9b95f7debbed.png)

Here we see that the fill is correctly very red, but the region on chr4 is not red as it should be. This is because the color is still green for some reason. We can see this when we zoom close into chr21.

At this point we get confused and wonder what is really up. If we go back to the plot where we don't overwrite the default color setting, and then inspect the data used by the using using `ggplot_build()`:

    #load the plot with both scales set in aes
    #get the actual data in the object
    gg_build = ggplot_build(gg)
    #inspect the 4th layer
    gg_build$data[[4]] %>% str

Out:

```
'data.frame':	140 obs. of  9 variables:
 $ fill    : chr  "#FB3400" "#FB3400" "#FB3400" "#FB3400" ...
 $ x       : num  -13.7 -13.7 -13.7 -13.6 -13.6 ...
 $ y       : num  14.8 14.8 14.9 14.9 14.9 ...
 $ group   : atomic  1 1 1 1 1 1 1 1 1 1 ...
  ..- attr(*, "n")= int 3
 $ PANEL   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ colour  : chr  "black" "black" "black" "black" ...
 $ size    : num  0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 ...
 $ linetype: num  1 1 1 1 1 1 1 1 1 1 ...
 $ alpha   : logi  NA NA NA NA NA NA ...
```

So there is indeed a hidden default `color = black`. If we use the same method on the plot where we set `color = NA`, we see:

```
'data.frame':	140 obs. of  9 variables:
 $ fill    : chr  "#FB3400" "#FB3400" "#FB3400" "#FB3400" ...
 $ x       : num  -13.7 -13.7 -13.7 -13.6 -13.6 ...
 $ y       : num  14.8 14.8 14.9 14.9 14.9 ...
 $ group   : atomic  1 1 1 1 1 1 1 1 1 1 ...
  ..- attr(*, "n")= int 3
 $ PANEL   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ colour  : logi  NA NA NA NA NA NA ...
 $ size    : num  0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 ...
 $ linetype: num  1 1 1 1 1 1 1 1 1 1 ...
 $ alpha   : logi  NA NA NA NA NA NA ...
```

So the NA value is still there, the `aes(color = score)` does not overwrite it as expected. The NA somehow gets converted to green, perhaps by using the last used color from the third layer?

Next, I tried setting `color = NULL`, but this gives an error due to length control:

    Error: Aesthetics must be either length 1 or the same as the data (140): colour

I tried setting the aes and color arguments of the scale in reverse order, but this had no effect when using NA and NULL still gave an error. In an attempt to be sneaky, I used `color = `, but this also did not work (length check fail).

So, I'm thinking there must be a bug somewhere in the detection for whether to overwrite the aes link with the default. I could not find an error in the "rect" method of the `layout_circle` in the source.

https://github.com/Bioconductor-mirror/ggbio/blob/master/R/layout_circle-method.R#L254

The handling of fill and color are identical, and the code is also identical to that for "bar" method. So I'm thinking the bug is futher up in either `parseArgsForAes` or `parseArgsForNonAes`?

**Comment by lawremi** [comment]
This is due to `aes()` renaming the /color/ aesthetic to /colour/ (an annoying aspect of ggplot). Version 1.21.5 should resolve this issue.

**Comment by Deleetdk** [comment]
I installed the latest version using `devtools::install_github` and it fixes my problem.

I looked over your commit and the changes seem reasonable, even if it is annoying that one has to deal with variant spellings. Maybe use regex? "colou?r" captures both.

I did think it may be a bug due to color/colour, so I tried using color/colour at various points in my code, but it didn't help.

Thanks for fixing this!
